### DISCUSSION TITLE
IDEA: Hive Blockchain extension

### DISCUSSION THREAD

**Discussion by Wend1go** [4ian/GDevelop#3320] [Enhancement ideas (and their technical discussions)]
I am currently working on an extension that gives access to the Hive blockchain.
Developers will be able to create a very straight forward login mechanism where players can authenticate themselves via their [Hive Keychain Wallet](https://hive-keychain.com/). 
It will be possible to send and receive Hive and Hive backed Dollars (HBD) crypto currencies or access information from the blockchain via [dHive](https://github.com/openhive-network/dhive). 
If there is demand, I could also add the possibility to create blog posts, add voting etc.

This extension would enable developers to create play2earn games or add in-app purchases in via the Hive crypto currency.
I already have a working prototype.

@4ian I have two questions:
- Am I guessing right that you would prefer it to become a standalone extension in [GDevelop-extensions](https://github.com/GDevelopApp/GDevelop-extensions)?
- For the example project, I was thinking about adding the possibility send tips in Hive/HBD cryptocurrency. If you are interested, you would need to create a Hive-Wallet for GDevelop. That could be another source of income for the project in the future.

What do you think?

**Comment by arthuro555** [comment]
Personally, I'd stay away from cryptocurrencies as they are pretty controversial and controversy could hurt GDevelop at this most critical stage of development where the GDevelop company is starting. I mean making a standalone extension for users that are into that is fine but a built-in feature or adding an integrated way to donate cryptocurrencies to GDevelop is IMO not a good move.

**Comment by Bouh** [comment]
I switch this issue to a discussion :)
I love crypto but it preferable as an user extension I guess because I agree with Arthuroo, we should think carefully about this.

**Reply by jjhesk** [reply]
extension is the best way to go and developers can build their wallets for any networks.  Eth, Binance, RSC, ethereum classic, celo, and more to do. Just like the metamask extensions that have common consensus for all the players.

**Comment by 4ian** [comment]
> Am I guessing right that you would prefer it to become a standalone extension in GDevelop-extensions?

Correct! This sounds like an excellent use case for a standalone extension.

> For the example project, I was thinking about adding the possibility send tips in Hive/HBD cryptocurrency. If you are interested, you would need to create a Hive-Wallet for GDevelop. That could be another source of income for the project in the future.

I'm happy to give it a try as I'm curious about this blockchain and potential applications - I never tried or learnt deeply about Hive before.

Concerning cryptocurrencies in general, I'm watching the space for now. It's both a very interesting but also hot and controversial topic.
I'm watching in particular projects that would be:
- really useful (not a get-rich-quick scheme), providing either authentication, monetization, item marketplaces or other form of social stuff to games,
- not harming the environment/sustainable in the long term
- accessible (ethereum gas fees for trading NFTs or creating new "smart contracts" are super super high, and more generally crypto stuff are super complex for users: you have to install all powerful extensions, you have not to lose your private keys, you have to aware of scams/rug pull).

I believe that a good project could provide incentive for creators to build and for players to come to these games, and we could help lower the entry barrier to these technologies with GDevelop - if they are following the criteria above :)
In the meantime, looking forward to see the extension and example! :)

**Comment by PascalLadalle** [comment]
I didn't know about the Hive blockchain, but as a crypto-enthusiast, I'm looking forward to giving it a try :)
I'm personally against NFTs, but they're all the rage now. Anything remotely related will attract _much_ more attention.
Keep up the good work! 🦾

**Comment by Wend1go** [comment]
I agree that crypto currencies are a little controversial. Especially Bitcoin gained a bad reputation in the public since it is often used as payment in combination with crypto trojans. But like many tools, it can be used for harming or for doing good.

I Picked the Hive blockchain because it is proof of stake instead of proof of work like Bitcoin. Instead of having thousands of computers race against each other for calculating new coins. The Hive chain has a builtin "savings" mechanism where you gain about 2% per year if you stake your hive ("Power up" in Hive speak). A lot better for the environment. There are also no gas fees for transactions. It uses a system of self replenishing resource credits that need to be paid per transaction to prevent spam. Powering up gives you more RC. But it is also possible to temporarily delegate Hive power to other wallets so that they are able to perform more actions on the chain per day.

As far as I know there is not a single game engine with builtin Hive functionality. This extension could indeed attract more users to GDevelop.

**Reply by 4ian** [reply]
Circling back to this, let me know if you need help or are still thinking of doing this!

**Comment by Silver-Streak** [comment]
I work in an role that involves a *lot* of people asking about crypto and blockchain, so I'm going to ask two questions that we must always ask whenever this gets brought up in any capacity before we end up doing any effort/research at my dayjob:

1. Is there anything functionally that this enables that existing technology cannot do today? (Despite the most common response: Immutable encrypted records are not exclusive nor unique to blockchain/crypto, nor do they inherently do anything unique with it other than being distributed encoding/encrypting/etc.)
2. Is there documented/tangible demand for this by people that would actually use it? (This means, excluding people seeking buzzwords, venture capitalists, or investors. Is there actual hard (or soft) numbers of gamers or developers that are seeking this functionality?)


The above isn't a "hey you shouldn't do this", as with any shiny and new tech, the above are just things you should consider before investing your own time on doing anything in this space, on top of the concerns 4ian already listed.

Blockchain is really cool tech, but I haven't found lots of instances where the technology actually adds any value to the people implementing it nor their users over existing/easier to maintain tech, so you have to be careful around it.  

(Unless it is NFT tech, which literally has no value other than the to the very small percentage of people that make up 90% of the transactions for them to inflate their value.)

**Reply by cian0** [reply]
Being in the Filipino play to earn scene for almost a year now, and speaking on behalf of a gaming guild of 63k members as project manager:
#1 major functionality is full ownership, you can take your nft assets anywhere and it could be used on different independent platforms/websites for:
a. Staking (like a time deposit) to get some yields in crypto token
b. Use it on a game as in game item that enables a certain feature
c. "Burning" or merging it with another NFT to produce a different/higher level one that unlocks more perks
d. Use it as an identification pass for events

#2 Usually, games that have play to earn mechanics have crypto tokens that can be earned using the NFTs, the real value usually is appreciated outside the platform. In case of Axie, there are a lot of establishments here in PH that accepts SLP token as payment, there was even one time where a car dealership marketed that they're accepting SLP as payment. For VCs of course they need first to understand the space to appreciate, in our guild's case, we almost have daily inquiries from investors on how they can invest and profit from the guild

I'd also like to add that I ran a small test business idea which was able to cater only 1k customers via FB, amount of transaction in just 2-3 months amounted to $2M usd from those few customers, maybe what I would like to say is that there's a vast blue ocean over there at crypto town where innovation is sought after and is greatly rewarded by the users. 

Don't shut down the idea, even just adding an optional extension that enables crypto transactions may flood your project with proposals and crazy investments like how we experienced after building an early p2e driven gaming guild. I don't think anyone has done this yet and I'm pretty sure other game engines would follow if they see someone succeeds in doing it.

If you want, we can even work together, our community would be happy to test/adopt the first few blockchain games that is developed by this awesome game engine.
